File: core/deterministic/src/motion_estimation/block.rs
pub struct Block {
    pub coord: (i32, i32, i32, i32),
    pub center: (i32, i32),
    pub mv: (i32, i32),
    pub mv_amp: f64,
}

impl Block {
    pub fn new(x: i32, y: i32, w: i32, h: i32) -> Block {
        Block {
            coord: (x, y, w, h),
            center: (x + w / 2, y + h / 2),
            mv: (0, 0),
            mv_amp: 0.0,
        }
    }

    pub fn check_inside_frame(&self, x: i32, y: i32, min: &(i32, i32, i32, i32), max: &(i32, i32, i32, i32)) -> bool {
        x >= min.0 && x <= max.0 && y >= min.1 && y <= max.1
    }

    pub fn calculate_mv_amp(&mut self, max_amp: &mut f64, min: &(i32, i32, i32, i32), max: &(i32, i32, i32, i32)) {
        let amp = ((self.mv.0.pow(2) + self.mv.1.pow(2)) as f64).sqrt();
        if amp > *max_amp {
            *max_amp = amp;
        }
        self.mv_amp = amp;
        if self.check_inside_frame(self.center.0, self.center.1, min, max) && self.mv_amp > *max_amp {
            *max_amp = self.mv_amp;
        }
    }
}

File: core/deterministic/python-core/src/motion_estimation/block.py
class Block():
    min = None
    max = None
    max_mv_amp = 0.0

    def __init__(self,x,y,w,h):
        self.coord   = (x,y,w,h)
        self.center  = (x+w//2,y+h//2)
        self.mv      = (0,0)
        self.mv_amp  = 0

    def check_inside_frame(self,x,y):
        """check if the searched box inside the target frame"""
        check = True
        if x<Block.min[0] or x>Block.max[0] \
            or y<Block.min[1] or y>Block.max[1]:
            check = False
        return check

    def calculate_mv_amp(self):
        """calculate L2 norm of motion-vector"""
        amp = (self.mv[0]**2 + self.mv[1]**2)**0.5
        #print(f"amp for block at {self.coord}: {amp}") 
        if amp > Block.max_mv_amp:
            Block.max_mv_amp = amp
        self.mv_amp = amp

    def __str__(self):
        return f"""
                Block: 
                coord={self.coord}, 
                center={self.center}, 
                mv={self.mv}, 
                mv_amp={self.mv_amp},
                min={Block.min},
                max={Block.max},
                max_mv_amp={Block.max_mv_amp}
        """
